test/test_pytree.py

```diff
@@ -163,24 +163,42 @@ def test_treemap(self):
         def run_test(pytree):
             def f(x):
                 return x * 3
-            sm1 = sum(map(tree_flatten(pytree)[0], f))
+            sm1 = list(map(f, tree_flatten(pytree)[0]))
             sm2 = tree_flatten(tree_map(f, pytree))[0]
             self.assertEqual(sm1, sm2)
 
             def invf(x):
                 return x // 3
+            sm3 = list(map(f, tree_flatten(pytree)[0]))
+            sm4 = tree_flatten(tree_map(f, pytree))[0]
+            self.assertEqual(sm3, sm4)
 
-            self.assertEqual(tree_flatten(tree_flatten(pytree, f), invf), pytree)
-
-            cases = [
-                [()],
-                ([],),
-                {'a': ()},
-                {'a': 1, 'b': [{'c': 2}]},
-                {'a': 0, 'b': [2, {'c': 3}, 4], 'c': (5, 6)},
-            ]
-            for case in cases:
-                run_test(case)
+        cases = [
+            [()],
+            ([],),
+            {'a': ()},
+            {'a': 1, 'b': [{'c': 2}]},
+            {'a': 0, 'b': [2, {'c': 3}, 4], 'c': (5, 6)},
+        ]
+        for case in cases:
+            run_test(case)
+
+
+    def test_treemap_multi_input(self):
```

Comment:
I think we can strengthen the test by

* Adding cases which handle not just 2 but maybe 3, 4 iterables and updating `f` accordingly.
* Having cases with dictionaries (not just lists).
* Having cases with interesting nested pytree like list of dictionary.

Wdyt?

Reply:
Interesting! have added more cases.
- a new function with 3 inputs
- iterating over tuples, lists, and dict together.